src/start_testbuilds.py

```diff
@@ -182,13 +192,32 @@ def main():
         LOGGER.info(f"Not triggering test jobs from boto3, as BUILD_CONTEXT is {build_context}")
         return
 
+    commit = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
+    build_framework = os.getenv("FRAMEWORK")
+
+    if config.is_deep_canary_mode_enabled() and (
```

Comment:
I think the main function is a bit of a clunky place to put this logic. Can you encapsulate this logic into a function and include a docstring as to why this is needed?

To make this extensible, I would write the function in a way that takes test_type as an argument rather than hard-coding conditions explicitly for deep canaries.

Reply:
The difficulty with taking `test_type` as an argument to another function is that it needs to be created by something when running with `deep_canary_mode = true` on the config file.

We need a branching path to start test jobs for deep-canary tests on PR anyway, because there are no images built on the build-job, and the `TEST_ENV_PATH` and `TEST_TYPE_IMAGES_PATH` files are never populated as a part of the build.

I can pull out the logic that exists within the if-condition into a separate helper, and write something like:
```
if config.is_deep_canary_mode_enabled():
    run_deep_canary_pr_testbuilds()
    return
```

Is that what you were envisioning?

Reply:
thanks for making the change!